src/core/etl/src/Flow/ETL/Row/Entry/StructureEntry.php

```diff
@@ -6,49 +6,47 @@
 
 use Flow\ArrayComparison\ArrayComparison;
 use Flow\ETL\Exception\InvalidArgumentException;
+use Flow\ETL\PHP\Type\Logical\StructureType;
 use Flow\ETL\Row\Entry;
 use Flow\ETL\Row\Reference;
 use Flow\ETL\Row\Schema\Definition;
 
 /**
- * @implements Entry<array<array-key, Entry>, array{name: string, entries: array<Entry>}>
+ * @implements Entry<array<array-key, mixed>, array{name: string, value: array<array-key, mixed>, type: StructureType}>
  */
 final class StructureEntry implements \Stringable, Entry
 {
     use EntryRef;
 
     /**
-     * @var array<Entry>
-     */
-    private readonly array $entries;
-
-    /**
+     * @param array<array-key, mixed> $value
+     *
      * @throws InvalidArgumentException
      */
-    public function __construct(private readonly string $name, Entry ...$entries)
-    {
+    public function __construct(
+        private readonly string $name,
+        private readonly array $value,
+        private readonly StructureType $type
+    ) {
         if ('' === $name) {
             throw InvalidArgumentException::because('Entry name cannot be empty');
         }
 
-        if (!\count($entries)) {
+        if (0 === \count($value)) {
             throw InvalidArgumentException::because('Structure must have at least one entry, ' . $name . ' got none.');
         }
 
-        $entryNames = \array_map(static fn (Entry $entry) => $entry->name(), $entries);
-
-        if (\count(\array_unique($entryNames)) !== \count($entryNames)) {
-            throw InvalidArgumentException::because('Each entry name in structure must be unique, given: ' . \implode(', ', $entryNames));
+        if (!$type->isValid($value)) {
+            throw InvalidArgumentException::because('Expected list of ' . $type->toString() . ' got different types.');
```

Comment:
Expected `structure` not `list`. Is StructureType also checking if there are no additional/missing fields in the values?